internals/overlord/servstate/manager_test.go

```diff
@@ -1661,6 +1653,44 @@ services:
 	c.Check(string(output), Equals, dir+"\n")
 }
 
+func (s *S) TestWaitDelay(c *C) {
+	s.setupEmptyServiceManager(c)
+
+	// Run the test binary with PEBBLE_TEST_WAITDELAY=1 (see TestMain).
+	testExecutable, err := os.Executable()
+	c.Assert(err, IsNil)
+	layer := parseLayer(c, 0, "layer", fmt.Sprintf(`
+services:
+    waitdelay:
+        override: replace
+        command: %s
+        environment:
+            PEBBLE_TEST_WAITDELAY: 1
+        kill-delay: 200ms
```

Comment:
this delay could be a lot shorter to speed up this test.

Reply:
Yeah, I think you're right -- no reason this has to be long. If the `WaitDelay` logic is not working, it'll fail hard even with a short timeout. Reduced to 10ms.